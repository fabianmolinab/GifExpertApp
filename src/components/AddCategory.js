import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

  const [inputValue, setInputValue] = useState('');

  // Cada vez que escriba
  const handleInputChange = (e) => {
    setInputValue(e.target.value)
  }

  // Cuando haces ENTER
  const handleSubmit = (e) => {
    e.preventDefault(); // Previene el refrescar de la pagina

    // ValidaciÃ³n del campo de busqueda
    if (inputValue.trim().length >= 2) {

      setCategories(cats => [inputValue, ...cats]);
      setInputValue('');
    }
  }
  return (
    <form onSubmit={handleSubmit}>
      <p> {inputValue} </p>
      <input
        type="text"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  )
};
AddCategory.propTypes = {

  setCategories: PropTypes.func.isRequired

};