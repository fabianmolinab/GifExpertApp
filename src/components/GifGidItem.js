import React from 'react';
import PropTypes from 'prop-types';

export const GifGidItem = ({title,url}) => {
  return (
    <div className="card animate__animated animate__wobble">
      
      <img 
        src={url} 
        alt={title}
      />
    </div>
  )
};

GifGidItem.propTypes = {

  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired

}

