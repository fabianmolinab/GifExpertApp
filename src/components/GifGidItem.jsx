import React from 'react'

export const GifGidItem = ({title,url}) => {
  return (
    <div className="card animate__animated animate__wobble">
      
      <img 
        src={url} 
        alt={title}
      />
    </div>
  )
}

