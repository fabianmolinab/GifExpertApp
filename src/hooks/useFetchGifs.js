import { useState, useEffect } from 'react'; 
import {getGifs} from '../helpers/getGifs';

export const useFetchGifs = (category) => {

  const [state, setState] = useState({  
    data: [],
    loading: true
  });

  useEffect( () => {
    getGifs(category) //Peticion a la API
      .then( imgs =>{
        setState({ 
          data: imgs,
          loading: false
        });          
      }) 
      .catch( error => console.log(error))
  }, [ category ]);
  
  return state; //{data: [], loading: true};
}