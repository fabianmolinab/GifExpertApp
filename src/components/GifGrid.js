import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGidItem } from './GifGidItem';


export const GifGrid = ({ category }) => {

  
  const {data:images, loading} = useFetchGifs( category ); // Llamado a la API  

  return (
    <>
      <h3> {category} </h3>

      { loading && <p>Cargando</p> } 

      <div className="card-grid">  
        {
          images.map( img => (
            <GifGidItem 
              key= { img } 
              { ...img }
            />
          ))
        }
      </div>
    </>
  )
};