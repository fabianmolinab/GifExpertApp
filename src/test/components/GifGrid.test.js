import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas de GifGrid', () => {

  const category = 'Voy';


  test('Debe mostrar el componente correctamente', () => {

    useFetchGifs.mockReturnValue({

      data: [],
      loading: true

    }); // Falcenando la data de el Hook

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  })

  test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {

    const gifs = [{
      id: 'Hola bebe',
      url: 'https://www.google.com',
      title: 'Cualquier cosa'
    }]
    
    useFetchGifs.mockReturnValue({

      data: gifs,
      loading: false

    }); // Falcenando la data de el Hook

    const wrapper = shallow(<GifGrid category={category} />);

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('p').exists() ).toBe(false);
    expect( wrapper.find('GifGidItem').length ).toBe( gifs.length );
  })
})