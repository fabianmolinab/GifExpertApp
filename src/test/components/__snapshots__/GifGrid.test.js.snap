// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifGrid Debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3>
     
    Voy
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGidItem
      id="Hola bebe"
      key="[object Object]"
      title="Cualquier cosa"
      url="https://www.google.com"
    />
  </div>
</Fragment>
`;

exports[`Pruebas de GifGrid Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3>
     
    Voy
     
  </h3>
  <p>
    Cargando
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
